XiEditor/Storage/LanguagesStore.swift

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+protocol LanguagesProvider {
+    func fetchRecentlyUsedLanguages() -> [String]
+    func saveRecentlyUsed(language: String)
+    var availableLanguages: [String] { get set }
```

Comment:
I'm not sure if we really want the `set` method exposed here? `availableLanguages` should only really be set in response to a core RPC, so exposing the setter to other users feels a bit iffy.

Reply:
@cmyr You are absolutely right. I think we need to change data flow a bit. I have some ideas and will share it soon.